### PR TITLE
FEAT: Support stable diffusion

### PR DESCRIPTION
**Pull request by codingl2k1**
- Enable self-host CI for stable diffusion

### CODE REVIEW

File: xinference/model/multimodal/core.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+from ...constants import XINFERENCE_CACHE_DIR
+from ..core import ModelDescription
+from .stable_diffusion.core import DiffusionModel
+
+MAX_ATTEMPTS = 3
+
+logger = logging.getLogger(__name__)
+
+
+class MultimodalModelSpec(BaseModel):
```
Comment (UranusSeven) [inline xinference/model/multimodal/core.py:30]: In `xinference`, a "model family" refers to a group of models sharing similar characteristics such as network structure. Each model within this family has its own detailed specification, termed "model spec."

Given this, perhaps a more suitable name for this class would be `MultimodalModelFamilyV1`. The `V1` suffix aids in differentiating potential incompatible versions.

File: xinference/client.py [hunk 2]
```diff
@@ -680,6 +713,8 @@ def get_model(self, model_uid: str) -> "ModelHandle":
                 raise ValueError(f"Unrecognized model ability: {desc['model_ability']}")
         elif desc["model_type"] == "embedding":
             return EmbeddingModelHandle(model_ref, self._isolation)
+        elif desc["model_type"] == "multimodal":
```
Comment (UranusSeven) [inline xinference/client.py:716]: `multimodal` includes text2img, img2text, text2audio, audio2text, etc. 

`images` could be a better name and is also the way OpenAI names their API.

File: xinference/model/multimodal/stable_diffusion/core.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                buffered = BytesIO()
+                _img.save(buffered, format="jpeg")
+                return base64.b64encode(buffered.getvalue())
+
+            with ThreadPoolExecutor() as executor:
+                results = list(map(partial(executor.submit, _gen_base64_image), images))
+                image_list = [Image(url=None, b64_json=s.result()) for s in results]
+            return ImageList(data=image_list)
+        else:
+            raise Exception(f"Unsupported response format: {response_format}")
```
Comment (UranusSeven) [inline xinference/model/multimodal/stable_diffusion/core.py:86]: How about raising a ValueError?

**Comment by UranusSeven** [comment]
A new RESTful API needs be added and it should follow the OpenAI's API spec: https://platform.openai.com/docs/api-reference/images/create

File: xinference/types.py [hunk 4]
```diff
@@ -17,6 +17,15 @@
 from typing_extensions import Literal, NotRequired, TypedDict
 
 
+class Image(TypedDict):
+    url: Optional[str]
+    b64_json: Optional[str]
+
+
+class ImageList(TypedDict):
+    data: List[Image]
```
Comment (UranusSeven) [inline xinference/types.py:26]: Besides `data`, there should be another field `created`.

Here's an example from OpenAI's doc:
```json
{
  "created": 1589478378,
  "data": [
    {
      "url": "https://..."
    },
    {
      "url": "https://..."
    }
  ]
}
```

File: xinference/client.py [hunk 5]
```diff
@@ -181,6 +182,38 @@ def chat(
         return self._isolation.call(coro)
 
 
+class MultimodalModelHandle(ModelHandle):
```
Comment (UranusSeven) [inline xinference/client.py:185]: A RESTful impl is also needed.

File: xinference/types.py (hunk 4 above)
Comment (codingl2k1) [reply xinference/types.py:27]: Fixed. Thanks.

File: xinference/client.py (hunk 2 above)
Comment (codingl2k1) [reply xinference/client.py:760]: Yes, multimodal includes text2img, img2text, text2audio, audio2text, etc. We can add additional methods to the `MultimodalModelHandle`.

File: xinference/model/multimodal/stable_diffusion/core.py (hunk 3 above)
Comment (codingl2k1) [reply xinference/model/multimodal/stable_diffusion/core.py]: Fixed. Thanks.

File: xinference/model/multimodal/core.py (hunk 1 above)
Comment (codingl2k1) [reply xinference/model/multimodal/core.py]: Fixed. Thanks.

File: xinference/client.py (hunk 5 above)
Comment (codingl2k1) [reply xinference/client.py:185]: Fixed. Thanks.

File: xinference/client.py (hunk 2 above)
Comment (codingl2k1) [reply xinference/client.py]: Fixed. Thanks.

**Review by UranusSeven — APPROVED** [review]

**Pull request merged**